### PR TITLE
feat: add `noNullableMisuseInRepositories()` rule test case

### PR DESCRIPTION
**Pull request by Charm-Xin**

### CODE REVIEW

**Comment by unlimitedsola** [comment]
let's rebase this

**Comment by Charm-Xin** [comment]
Yes, I know.

**Comment by unlimitedsola** [comment]
> Yes, I know.

This still doesn't look right, it contains duplicated changes from our previous PR.

**Comment by Charm-Xin** [comment]
Is there a possibility that I write the same commit title, but the modified content inside is different?

**Comment by Charm-Xin** [comment]
> This still doesn't look right, it contains duplicated changes from our previous PR.

Is there a possibility that I write the same commit title, but the modified content inside is different?

**Comment by Charm-Xin** [comment]
> let's rebase this

My understanding is that rebase `using-spring-nullable-test-case` onto `using-spring-nullable`. Did I misunderstand?

**Comment by unlimitedsola** [comment]
> > let's rebase this
> 
> My understanding is that rebase `using-spring-nullable-test-case` onto `using-spring-nullable`. Did I misunderstand?

You should always rebase onto the main branch, as your PR should always base on that branch.

File: src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java
Comment (unlimitedsola) [inline src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java:1]: Do we really need entity definitions? As we are not spinning up a real Spring application, I think we can use simple placeholders for brevity

Comment (Charm-Xin) [reply src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java:1]: Oh, indeed.

File: src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java [hunk 1]
```diff
@@ -1,102 +1,31 @@
...
-            return test1;
-        }
-
-        public String getTest2() {
-            return test2;
-        }
-    }
-
-    public interface CompliantRepo extends JpaRepository<EntityCompliant, Long> {
+    public interface CompliantRepo extends JpaRepository<Object, Long> {
```
Comment (unlimitedsola) [inline src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java]:
```suggestion
    public interface Compliant extends JpaRepository<Object, Long> {
```
Let's be consistent with our namings

File: src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java [hunk 2]
```diff
@@ -5,23 +5,23 @@
...
         Object findObjectBy(@org.springframework.lang.Nullable String test);
     }
 
-    public interface NonCompliantRepo1 extends JpaRepository<Object, Long> {
+    public interface NonCompliant1 extends JpaRepository<Object, Long> {
 
         @jakarta.annotation.Nullable
         Object findBy(String test);
 
         Object findObjectBy(@jakarta.annotation.Nullable String test);
```
Comment (unlimitedsola) [inline src/test/java/dev/ocpd/jsensible/rules/spring/NoNullableMisuseInRepositories.java]: I suggest we split this into two as we need both of them to be failing, if only one fails the test will count as a pass which is an issue.

**Review by unlimitedsola — APPROVED** [review]

**Pull request merged**